enterprise/server/ociregistry/ociregistry.go

```diff
@@ -410,17 +412,23 @@ func fetchBlobOrManifestFromCache(ctx context.Context, w http.ResponseWriter, bs
 	w.Header().Add(headerContentType, blobMetadata.GetContentType())
 	w.WriteHeader(http.StatusOK)
 
-	if writeBody {
-		blobRN := digest.NewCASResourceName(
-			blobCASDigest,
-			"",
-			repb.DigestFunction_SHA256,
-		)
-		blobRN.SetCompressor(repb.Compressor_ZSTD)
-		return cachetools.GetBlob(ctx, bsClient, blobRN, w)
-	} else {
+	if !writeBody {
 		return nil
 	}
+	blobRN := digest.NewCASResourceName(
+		blobCASDigest,
+		"",
+		repb.DigestFunction_SHA256,
+	)
+	blobRN.SetCompressor(repb.Compressor_ZSTD)
+	err = cachetools.GetBlob(ctx, bsClient, blobRN, w)
```

Comment:
if you want this to work even if there is an error part of the way through, you can use `ioutil.Counter` from `server/util/ioutil` and `io.MultiWriter` to write to both `w` and the counter, then report `counter.Count()` unconditionally.

Reply:
Oh neat, I hoped that existed!